src/logic/safe/store/reducer/gatewayTransactions.ts

```diff
@@ -122,15 +136,30 @@ export const gatewayTransactionsReducer = handleActions<GatewayTransactionsState
         }
 
         const newTx = value.transaction
+
         if (label === 'queued') {
           if (newQueued?.[txNonce]) {
-            newQueued[txNonce] = [...newQueued[txNonce], newTx]
+            const txIndex = newQueued[txNonce].findIndex(({ id }) => sameString(id, newTx.id))
+
+            if (txIndex !== -1) {
+              const storedTx = newQueued[txNonce][txIndex]
+              newQueued[txNonce][txIndex] = getUpdatedTx(storedTx, newTx)
```

Comment:
Why are we not simply overwriting the existing value like we do for history transactions? i.e. `newQueued[txNonce][txIndex] = newTx`

Reply:
We want to preserve `txDetails`. Perhaps we could just `merge({}, storedTx, newTx)` by default. I will look into it.